src/ReflectionFileNamespace.php

```diff
@@ -83,6 +83,9 @@ class ReflectionFileNamespace
      */
     public function __construct($fileName, $namespaceName, Namespace_ $namespaceNode = null)
     {
+        if (!is_string($fileName)) {
```

Comment:
1 - keep.

If we try to imagine real life examples, then `ReflectionFileNamespace` class would only be created once per namespace. And I haven't seen any codebases, where unique namespace count is fare superior (or even equal) to class/interface/trait count in these namespaces.